ambry-store/src/main/java/com.github.ambry.store/StoreFindToken.java

```diff
@@ -78,13 +101,18 @@ private StoreFindToken(Type type, Offset offset, StoreKey key, UUID sessionId) {
     this.offset = offset;
     this.storeKey = key;
     this.sessionId = sessionId;
+    this.incarnationId = incarnationId;
     this.bytesRead = -1;
   }
 
   void setBytesRead(long bytesRead) {
     this.bytesRead = bytesRead;
   }
 
+  void setInclusive() {
```

Comment:
I do like to minimize the number of `StoreFindToken` constructors - but would like one constructor that takes in every parameter including whether the token is inclusive/exclusive. This should be set at construction time and not after the fact. So I would advise against a `setInclusive()` method.